### PR TITLE
Issue1677

### PR DESCRIPTION
**Pull request by tonypark0403**
When a file is added, the default option of AUTO is enable but not showing checking mark.

### CODE REVIEW

**Comment by humphd** [comment]
This change is too extreme, in that it will cause the function to always force a reload.  What you need to do is add a `true` to the caller in the particular spot where navigation is happening, which I linked to in the previous discussion.

**Comment by tonypark0403** [comment]
I changed the file as below.
- force = true
+ reload(true)
But, I couldn't commit it again because of the error message.
git push -u upstream issue1677
remote: Permission to mozilla/brackets.git denied to tonypark0403.
fatal: unable to access 'https://github.com/mozilla/brackets.git/': The requested URL returned error: 403

**Comment by humphd** [comment]
You can't push to Mozilla's upstream repo, since you don't have write permissions.  You need to push to your origin, and it will automatically add it to this pull request.

**Comment by tonypark0403** [comment]
I did it. Thank you very much.

**Comment by humphd** [comment]
The code you have now will do the exact same thing as before, just indirectly: you're checking if `force` is null, and if so calling the *same* function again with `true`.  That's not what I'm suggesting you do.

Instead, I want you to change the code as described here: https://github.com/mozilla/thimble.mozilla.org/issues/1677#issuecomment-276840542.  Specifically, change https://github.com/mozilla/brackets/blob/master/src/extensions/default/bramble/lib/PostMessageTransport.js#L164 to use `reload(true);`.

**Comment by tonypark0403** [comment]
Thank you for helping to find out what you want me to do.
But, during the git processing, I got the non-fast-forward problem.
Can you  delete my previous pull request or please let me know other way in this case.

$ git push origin issue1677
To https://github.com/tonypark0403/brackets.git
! [rejected]            issue1677 -> issue1677 (non-fast-forward)
error: failed to push some refs to 'https://github.com/tonypark0403/brackets.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.

**Comment by humphd** [comment]
You can make that push work by doing `git push origin issue1677 -f` (note the `-f`, which will force it to overwrite what's already there).  Make sure this is what you want, though, since it *will* overwrite the commit(s) on Github.

**Comment by tonypark0403** [comment]
I recommit it again after fixing it by your instruction.
But, even though the commit is finished, it didn't show anything on mozilla/brackets.

**Comment by tonypark0403** [comment]
amended by request

**Comment by gideonthomas** [comment]
@humphd I'm not sure when the `Page.reload` event is triggered. But shouldn't we force reload only for `Page.navigate` events?

**Comment by humphd** [comment]
Yes, only for `Page.navigate`.

File: src/extensions/default/bramble/lib/PostMessageTransport.js [hunk 1]
```diff
@@ -161,7 +161,7 @@ define(function (require, exports, module) {
         // transport, check message before sending to remote, and reload if necessary
         // without actually sending to remote for processing.
         if(msg.method === "Page.reload" || msg.method === "Page.navigate") {
-            reload();
+            reload(true);
```
Comment (gideonthomas) [inline src/extensions/default/bramble/lib/PostMessageTransport.js]: We should force reload (i.e. send `true` to the `reload` function) only for `Page.navigate` not for `Page.reload`

File: src/extensions/default/bramble/lib/PostMessageTransport.js [hunk 2]
```diff
@@ -160,10 +160,13 @@ define(function (require, exports, module) {
...
         // without actually sending to remote for processing.
-        if(msg.method === "Page.reload" || msg.method === "Page.navigate") {
+        if(msg.method === "Page.reload") {
             reload();
             return;
-        }
+        } else if(msg.method === "Page.navigate") {
+            reload(true);
+            return;
+	}
```
Comment (gideonthomas) [inline src/extensions/default/bramble/lib/PostMessageTransport.js]: let's use a 4-space indent here instead of a tab.

**Comment by gideonthomas** [comment]
Thanks for the contribution @tonypark0403!

**Pull request merged**